models/collectionswap/ethereum/collectionswap_ethereum_events.sql

```diff
@@ -0,0 +1,163 @@
+{{ config(
+        schema='collectionswap_ethereum',
+        alias = 'events',
+        partition_by = ['block_date'],
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['block_date', 'unique_trade_id']
+        )
+}}
+
+{%- set project_start_date = '2023-03-29' %}
+
+WITH
+raw_trades as (
+    select *
+    , row_number() over (partition by tx_hash order by evt_index asc, sub_order_id asc) as sub_tx_id
+    from(
+        select
+            block_number, block_time, evt_index, tx_hash, buyer, seller,
+            posexplode(nft_id_array) as (sub_order_id, token_id),
+            amount_raw/number_of_items as amount_raw,
+            platform_fee_amount_raw/number_of_items as platform_fee_amount_raw,
+            royalty_fee_amount_raw/number_of_items as royalty_fee_amount_raw,
+            trade_fee_amount_raw/number_of_items as trade_fee_amount_raw,
+            royalty_fee_receive_address,
+            project_contract_address,
+            number_of_items,
+            CASE WHEN number_of_items > 1 THEN 'Bundle Trade'
+                    ELSE 'Single Item Trade'
+                   END as trade_type,
+            trade_category
+        from(
+            select
+                 evt_block_number as block_number
+                ,evt_block_time as block_time
+                ,evt_index
+                ,evt_tx_hash as tx_hash
+                ,null as buyer
+                ,contract_address as seller
+                ,'Buy' as trade_category
+                ,nftIds as nft_id_array
+                ,cardinality(nftIds) as number_of_items
+                ,cast(outputAmount as decimal(38)) as amount_raw
+                ,cast(protocolFee as decimal(38)) as platform_fee_amount_raw
+                ,get_json_object(royaltyDue[0], '$.amount') as royalty_fee_amount_raw
+                ,get_json_object(royaltyDue[0], '$.recipient') as royalty_fee_receive_address
+                ,cast(tradeFee as decimal(38)) as trade_fee_amount_raw
+                ,contract_address as project_contract_address
+            from collectionswap_ethereum.CollectionPool_evt_SwapNFTOutPool e
```

Comment:
jinja

Reply:
oh wow, totally missed these and expected the pre-commit to catch them. :thinking: